From patchwork Wed Oct 20 13:00:22 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Kyle Copperfield <kmcopper@danwin1210.me>
X-Patchwork-Id: 12572363
Return-Path: <linux-bluetooth-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 84AE3C433F5
	for <linux-bluetooth@archiver.kernel.org>;
 Wed, 20 Oct 2021 13:07:59 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 66565601FF
	for <linux-bluetooth@archiver.kernel.org>;
 Wed, 20 Oct 2021 13:07:59 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S230186AbhJTNKM (ORCPT
        <rfc822;linux-bluetooth@archiver.kernel.org>);
        Wed, 20 Oct 2021 09:10:12 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:60884 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S229911AbhJTNKL (ORCPT
        <rfc822;linux-bluetooth@vger.kernel.org>);
        Wed, 20 Oct 2021 09:10:11 -0400
X-Greylist: delayed 384 seconds by postgrey-1.37 at lindbergh.monkeyblade.net;
 Wed, 20 Oct 2021 06:07:56 PDT
Received: from danwin1210.me (danwin1210.me [IPv6:2a01:4f8:c010:d56::1])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 9C46DC06161C;
        Wed, 20 Oct 2021 06:07:56 -0700 (PDT)
Received: from danwin1210.me (unknown [10.9.0.3])
        (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
         key-exchange X448 server-signature ECDSA (P-384) server-digest SHA384
         client-signature ED448)
        (Client CN "danwin1210.me", Issuer "danwin1210.me" (verified OK))
        by mail.danwin1210.me (Postfix) with ESMTPS id 22FFF1F43B;
        Wed, 20 Oct 2021 13:01:27 +0000 (UTC)
Received: from prine.. (unknown [10.9.0.1])
        (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
         key-exchange X25519 server-signature ECDSA (P-384) server-digest
 SHA384)
        (No client certificate requested)
        by danwin1210.me (Postfix) with ESMTPSA id 4CE635A644;
        Wed, 20 Oct 2021 13:00:33 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=danwin1210.me;
        s=mail; t=1634734856;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding;
        bh=wvlKkaYQfmgd1AQPtmZAcpJC7xRHMNlCP4y6n6AzJL0=;
        b=A42M3zm3idaqWTNZEkfWhmyNrLjK3eEm2rVVigtKnLytKIbGAA9tFb6l793sC/yTCg/Dse
        Y1iuZ3oyAGgqXqqz9LUFjjgfouuoYZ9u0V1X+QcNHRqNUWya9zvIiO/Eaire0ErLG9md0j
        8w3+fYkuB1wSGfmy2L8sYT3Oppt364JhQ8QoaEdqQ7NewTRlRInVeep03oPJSxoXOs3b0i
        gKbTZOD1BTcNPZidfMB2qcn0Khxcer4pn0GDpc8WqTc0/NzWV1guwrsjH5cyxH4Ec0xQ/Q
        fS9sgTCBMq0qKDnPUhwvSz1Xy6KsrdaneqU7xkg6TCkHfSCzx/pihukz3rtPX+8P6Q+SCD
        +Heva/W/Mk/H978E3DzFo5sx3sYjMe8BUsI6P2XKos7VwNo8KYogc6PAXzLNnkTVdqd2N1
        D7L2s5os1u7ooSmvhGJOGWCTSnaVc8qkV8YxfO4RqMxm/Mz81kKJBJXuBFVurXf3j5w3Ei
        4pVzZAzcXFXGChpsqH369q+j98BO2hSX9zVExoSbjRZDVdX3wO3hNB7juLt+hQMVkD5XTw
        oIo5J8cfJhyim8rJHWXy8AxgCsTXojr8SD3c1rkfYLLo4mjInndjBwrMkDeUDJEVps+Vxq
        bXxHP62fWJgTcOx64bBJBNo2ChbtN7EIgcCYtWZK16A+9LX6OeHGU=
DKIM-Signature: v=1; a=ed25519-sha256; c=relaxed/relaxed; d=danwin1210.me;
        s=ed25519; t=1634734856;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding;
        bh=wvlKkaYQfmgd1AQPtmZAcpJC7xRHMNlCP4y6n6AzJL0=;
        b=O8J1B6BfYwKsfAF4l9yz6jR1z2rbSQ8qXrTbVJw0cc8NpkzyNa91JHvhmHKe6iNcEB7bvT
        bfcyMKJYcXIiS4AQ==
From: Kyle Copperfield <kmcopper@danwin1210.me>
To: Marcel Holtmann <marcel@holtmann.org>,
        Johan Hedberg <johan.hedberg@gmail.com>,
        Luiz Augusto von Dentz <luiz.dentz@gmail.com>
Cc: Kyle Copperfield <kmcopper@danwin1210.me>,
        Dragan Simic <dragan.simic@gmail.com>,
        linux-bluetooth@vger.kernel.org, linux-kernel@vger.kernel.org
Subject: [PATCH] Bluetooth: btsdio: Do not bind to non-removable BCM4345 and
 BCM43455
Date: Wed, 20 Oct 2021 13:00:22 +0000
Message-Id: <20211020130023.196651-1-kmcopper@danwin1210.me>
MIME-Version: 1.0
Precedence: bulk
List-ID: <linux-bluetooth.vger.kernel.org>
X-Mailing-List: linux-bluetooth@vger.kernel.org

BCM4345 and BCM43455 devices soldered onto the PCB (non-removable),
use an UART connection for Bluetooth, such as in the Pinebook Pro,
and the advertised btsdio support as an SDIO function is ignored.

Reviewed-by: Dragan Simic <dragan.simic@gmail.com>
Signed-off-by: Kyle Copperfield <kmcopper@danwin1210.me>
---
 drivers/bluetooth/btsdio.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/bluetooth/btsdio.c b/drivers/bluetooth/btsdio.c
index 199e8f7d426d..795be33f2892 100644
--- a/drivers/bluetooth/btsdio.c
+++ b/drivers/bluetooth/btsdio.c
@@ -295,6 +295,8 @@ static int btsdio_probe(struct sdio_func *func,
 		switch (func->device) {
 		case SDIO_DEVICE_ID_BROADCOM_43341:
 		case SDIO_DEVICE_ID_BROADCOM_43430:
+		case SDIO_DEVICE_ID_BROADCOM_4345:
+		case SDIO_DEVICE_ID_BROADCOM_43455:
 		case SDIO_DEVICE_ID_BROADCOM_4356:
 			return -ENODEV;
 		}
